File: problems/3019-number-of-changing-keys/solution.java
class Solution {
    public int countKeyChanges(String s) {
        int count = 0;
        if (s.length() == 0) {
            return 0;
        }
        char prev = Character.toLowerCase(s.charAt(0));
        for (int i = 1; i < s.length(); i++) {
            char curr = Character.toLowerCase(s.charAt(i));
            if (curr != prev) {
                count++;
                prev = curr;
            }
        }
        return count;
    }
}

File: problems/3019-number-of-changing-keys/solution.c
int countKeyChanges(char* s) {
    int changes = 0;
    int len = 0;
    while (s[len] != '\0') {
        len++;
    }
    if (len <= 1) {
        return 0;
    }

    for (int i = 1; i < len; i++) {
        char prev = s[i - 1];
        char curr = s[i];

        if (prev >= 'A' && prev <= 'Z') {
            prev = prev + 32;
        }
        if (curr >= 'A' && curr <= 'Z') {
            curr = curr + 32;
        }

        if (prev != curr) {
            changes++;
        }
    }

    return changes;
}